lib/hl7/parser.ex

```diff
@@ -0,0 +1,118 @@
+defmodule HL7.Parser do
+  @carriage_return 4
+  @field 3
+  @repetition 2
+  @component 1
+  @sub_component 0
+
+  def parse(<<"MSH|^~\\&", rest::binary()>> = text) do
```

Comment:
Are there any tests that call this function?

Reply:
I'm writing some. Maybe I'll PR them to your branch.

Reply:
It got test indirectly (as almost every existing test went through this path now....) but I added an explicit test that comparse the list parse of the original code and this new code to verify that they are identical.